packages/core/src/state-management/repository.ts

```diff
@@ -97,7 +103,9 @@ export class Repository {
       deps.conflictResolution,
       this.logger,
       (streamId) => this.fromMemoryOrStore(streamId),
-      (streamId, opts) => this.load(streamId, opts)
+      (streamId, opts) => this.load(streamId, opts),
+      this.indexStreamIfNeeded,
```

Comment:
I want to moving out of the habit of the state manager calling back up into the Repository.  That is a circular dependency nightmare IMO.  This seems to be making the problem worse, not better

Reply:
done.

Reply:
So I guess the plan is to fix this in the follow-up PR?  Can you add a TODO comment?

Reply:
Yes, adding the TODOs right now to close this one out.